### ISSUE TITLE
Align APIs around Global State

### ISSUE THREAD

**Issue by tobias-tengler** [ChilliCream/hotchocolate#4188]
### Is your feature request related to a problem?

Currently there are three main APIs around the "Global State" concept, and all are named differently.

`IResolverContext.ContextData["name"]` for Code-first
`[GlobalState("name")]` for Annotation-based
`IQueryRequestBuilder.SetProperty("name", value)` for Interceptors

This is really confusing in my opinion. 

### The solution you'd like

The same name should be used for all three APIs, something like:

`IResolverContext.GetGlobalState<T>("name")` or `IResolverContext.GlobalState["name"]`
`[GlobalState("name")]`
`IQueryRequestBuilder.SetGlobalState("name", value)`

### Product

Hot Chocolate

**Comment by PascalSenn** [comment]
Hm.. If we think about renaming this, we should also consider this:
```csharp
IResolverContext.GetRequestState<T>("name") or IResolverContext.RequestState["name"]
[RequestState("name")]
IQueryRequestBuilder.SetRequestState("name", value)
```

Because Global state does not really tell what it is. Schema.ContextData, Request.ContextData or Application Level Global state?

**Comment by michaelstaib** [comment]
This is a more difficult thing :) will have to think about.

**Comment by tobias-tengler** [comment]
Closed in favor of #4494
